File: app/src/main/java/com/infy/dagger2implementation/di/module/ActivityModule.java
package com.infy.dagger2implementation.di.module;

import android.app.Activity;
import android.support.v4.app.Fragment;

import com.infy.dagger2implementation.activity.MainActivity;
import com.infy.dagger2implementation.base.BaseActivity;
import com.infy.dagger2implementation.base.BaseFragment;
import com.infy.dagger2implementation.di.qualifier.ActivityContext;

import dagger.Binds;
import dagger.Module;

@Module
public abstract class  ActivityModule{
    @Binds
    @ActivityContext
    abstract Activity provideActivity(BaseActivity baseActivity);
    @Binds
    abstract Fragment provideBaseFragment(BaseFragment baseFragment);
    @Binds
    @ActivityContext
    abstract Activity provideMainActivity(MainActivity mainActivity);

}

File: app/src/main/java/com/infy/dagger2kotlin/di/module/ActivityModule.kt
package com.infy.dagger2kotlin.di.module

import android.app.Activity
import android.support.v4.app.Fragment
import com.infy.dagger2kotlin.activity.MainActivity
import com.infy.dagger2kotlin.base.BaseActivity
import com.infy.dagger2kotlin.base.BaseFragment
import com.infy.dagger2kotlin.di.qualifier.ActivityContext
import dagger.Binds
import dagger.Module

@Module
abstract class ActivityModule {
    @Binds
    @ActivityContext
     abstract fun provideActivity(baseActivity: BaseActivity): Activity

    @Binds
     abstract fun provideBaseFragment(baseFragment: BaseFragment): Fragment

    @Binds
    @ActivityContext
     abstract fun provideMainActivity(mainActivity: MainActivity): Activity

}